pkg/apis/operator/v1alpha1/types_garden.go

```diff
@@ -253,6 +253,17 @@ type DNS struct {
 	// +kubebuilder:validation:MinItems=1
 	// +optional
 	Domains []string `json:"domains,omitempty"`
+	// Provider is a provider name. This field is immutable.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Provider is immutable"
+	// +optional
+	Provider *string `json:"provider,omitempty"`
+	// ProviderConfig is the provider-specific configuration passed to DNSRecord resource.
+	// +optional
+	ProviderConfig *runtime.RawExtension `json:"providerConfig,omitempty"`
+	// SecretRef is a reference to a Secret object containing the DNS provider credentials where the records are
+	// managed.
+	// +optional
+	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
```

Comment:
Do we want to keep the provider configuration in the `virtualCluster` section only? It's kind of unexpected that this provider is also used to create wildcard ingress entries.

Any thoughts (cc @rfranzke)?

Also, I think we need more validation to ensure the `SecretRef` is given if `Provider` is configured.

Reply:
For now, we have this `DNS` struct in the `virtualCluster` section, which is not optimal, but just a "cosmetic" issue, I guess. What do you suggest @timuthy?

Reply:
We discussed some options today with @MartinWeindel and @oliver-goetz, mainly to allow different provider for the different domains you can configure. Martin will push a new commit, so we can continue the discussion based on those changes.

Reply:
The two `domains` fields now have the type `JSON` instead of `string` to support both specifying just a string or a JSON object with fields `name` and `provider`.

See commit [Support separate DNS providers per domain name](https://github.com/gardener/gardener/pull/10645/commits/36c6675b01282c65afe41b4220d75aae03c8ea7f#diff-ea71dc311fcc2168c86ebf4181eae329381c0f364225446177875166d7dd62ac)